src/domain/Game.ts

```diff
@@ -39,4 +42,85 @@ export class Game {
     getWordBeingGuessed() : string {
         return this.wordBeingGuessed;
     }
+
+    isGameOver() : string {
+        const maxMissedGuesses = 10;
+        const correctGuessesRequiredToWin = this.getUniqueLettersInAWord();
+
+        if (this.getMissedGuesses() == maxMissedGuesses)
+            return "Game is lost";
+        else if (this.getCorrectGuesses() == correctGuessesRequiredToWin)
+            return "Game is won";
+        else
+            return "Game is still in progress";
+    }
+
+    private getUniqueLettersInAWord() : number {
+        const word = this.getWordBeingGuessed();
+        let seenLetters = new Array();
+
+        for (let i = 0; i < word.length; i++) {
```

Comment:
jeigu ieškai unikalių raidžių tai tau Set struktūra šitą sprendžia, padaryk:
1. const wordLetters = Array.from(word);
2. return new Set(lettersArray).length;